connect/config.py

```diff
@@ -62,6 +64,9 @@ def __init__(
             if isinstance(products, str) \
             else products or []
 
+        # Store last created instance
+        Config.instance = self
```

Comment:
You can actually create two config instances. That part is related to my previous comment:

> Is the case with multiple configs contemplated as a use case? In that case it should be decided whether we want Config.instance to contain the first initialized configuration or the last.

In this case, when a Config instance is created, the global instance points to it. If you want to have multiple configs, you have to store the instance yourself (for example sending it to the resource that needs to use it). If you access the global instance, due to that line you would get the last config that has been parsed. This differs from how it worked before, where calling Config(...) only sets the new config if it had not been set before. That's why I asked this, if we want to only update the global instance if it has not been set before, it is a simple change. Please confirm what's the preferred behaviour.

Reply:
instead of having first and others instances - better to have class-methods 
get_instance set_instance - to manage default instance?

"first" approach is kind of ok also, but ...

Javi - what you think?

Reply:
It depends on whether we need these scenarios:

* Change the configuration after it has been initialized.
* Have multiple config objects.

If we don't need these, I would go for the immutable singleton approach, which has get_instance() but no set_instance(). The config is still global as you want, but it is immutable once initialized as I'd like.